html/ops/generate_account_ownership_keys.php

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env php
+<?php
+
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2008 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/user.inc");
+require_once("../inc/util.inc");
+
+$config = get_config();
+$keydir = parse_config($config, "<key_dir>");
+$private_key_path = "file://$keydir/account_ownership_private.pem";
+$public_key_path = "file://$keydir/account_ownership_public.pem";
+
+if (php_sapi_name() == "cli") {
+    /*
+      Running via the command line (or in makeproject)
+      We'll have secure access to the keys folder
+    */
+    if (!empty($argv[1])) {
+      if ($argv[1] == "overwrite") {
+        if (file_exists($private_key_path)) {
+          // If the private key exists, delete it.
+          unlink("$keydir/account_ownership_private.pem");
+          echo "erased 'account_ownership_private.pem' \n";
+        }
+        if (file_exists($public_key_path)) {
+          // If the public key exists, delete it.
+          unlink("$keydir/account_ownership_public.pem");
+          echo "erased 'account_ownership_public.pem' \n";
+        }
+      }
+    }
+
+    if ((!file_exists($private_key_path)) && (!file_exists($public_key_path))) {
+      $generated_pkey = openssl_pkey_new(array(
+        'digest_alg' => 'sha512',
+      	'private_key_bits' => 4096,
+      	'private_key_type' => OPENSSL_KEYTYPE_RSA
+      ));
+
+      $pubkey = openssl_pkey_get_details($generated_pkey); // Get the public key from the generated pkey pair
+      file_put_contents("$keydir/account_ownership_public.pem", $pubkey['key']); // Save the public key to disk
+      openssl_pkey_export_to_file($generated_pkey, "$keydir/account_ownership_private.pem"); // Save the private key to disk
+      openssl_pkey_free($generated_pkey); // Free key data securely from memory
+      echo "Generated a new account ownership keypair.";
```

Comment:
You should check that there wasn't an error return code from the previous steps before echoing this success message.